Sources/MongoSwift/BSON/BSONDecoder.swift

```diff
@@ -345,7 +345,7 @@ extension _BSONDecoder {
             return Date(timeIntervalSince1970: seconds)
         case .iso8601:
             guard #available(macOS 10.12, iOS 10.0, watchOS 3.0, tvOS 10.0, *) else {
-                throw MongoError.bsonDecodeError(message: "ISO8601DateFormatter is unavailable on this platform.")
+                fatalError("ISO8601DateFormatter is unavailable on this platform.")
```

Comment:
should we ever even get here? Shouldn't we be preventing choosing this value in the first place on this platform?

Reply:
I don't believe we ever can get there, but the compiler doesn't know that. It needs to have the `guard` to know it's in a context where that availability holds, despite the fact that the case could only be evaluated in such a context. Without the `guard`, compilation fails.

Reply:
oh I would have thought `#available` worked like a prepocessor macro like `#ifdef`, so the option wouldn't even be seen in this context. Cool, thanks for clearing this up.